eio/h2_eio.ml

```diff
@@ -35,14 +35,17 @@ module Server = struct
       ?(config = H2.Config.default)
       ~request_handler
       ~error_handler
+      ~sw
       client_addr
       socket
     =
     let connection =
       H2.Server_connection.create
         ~config
         ~error_handler:(error_handler client_addr)
-        (request_handler client_addr)
+        (fun reqd ->
+          Eio.Fiber.fork ~sw (fun () ->
+            request_handler client_addr reqd))
```

Comment:
Going by the [fork docs](https://ocaml-multicore.github.io/eio/eio/Eio/Fiber/index.html#val-fork). 
This should run the `request_handler` immediately, without switching to any other fiber first, and then schedule the calling Fiber (h2) next. 

> A fiber runs until it performs an IO operation (directly or indirectly). At that point, it may be suspended and the next fiber on the run queue runs.

From the Eio docs on Fibers, I expect the handler to run to a completion IO operation or `yield` which makes sense to me.

Reply:
We should actually close this PR.

The reasoning:
- this is trivial to do in user land
- forking at the library level disallows blocking workflows, which might be desirable to some users